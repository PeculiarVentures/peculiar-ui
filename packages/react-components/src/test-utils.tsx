import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
