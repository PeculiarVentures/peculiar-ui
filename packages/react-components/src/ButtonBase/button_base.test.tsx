import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { ButtonBase } from './index';

describe('<ButtonBase />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<ButtonBase>Text</ButtonBase>);

    expect(asFragment()).toMatchSnapshot();
  });
});
