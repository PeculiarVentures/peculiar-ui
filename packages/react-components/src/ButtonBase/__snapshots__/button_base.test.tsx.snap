// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBase /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-black);
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:not(:disabled):focus {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-5);
}

.emotion-0:disabled {
  color: var(--pv-color-gray-7);
}

.emotion-1 {
  margin: 0;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  color: inherit;
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Text
    </span>
  </button>
</DocumentFragment>
`;
